src/client/client-host-ref.ts

```diff
@@ -3,41 +3,35 @@ import { reWireGetterSetter } from '@utils/es2022-rewire-class-members';
 
 import type * as d from '../declarations';
 
-/**
- * A WeakMap mapping runtime component references to their corresponding host reference
- * instances.
- *
- * **Note**: If we're in an HMR context we need to store a reference to this
- * value on `window` in order to maintain the mapping of {@link d.RuntimeRef}
- * to {@link d.HostRef} across HMR updates.
- *
- * This is necessary because when HMR updates for a component are processed by
- * the browser-side dev server client the JS bundle for that component is
- * re-fetched. Since the module containing {@link hostRefs} is included in
- * that bundle, if we do not store a reference to it the new iteration of the
- * component will not have access to the previous hostRef map, leading to a
- * bug where the new version of the component cannot properly initialize.
- */
-const hostRefs: WeakMap<d.RuntimeRef, d.HostRef> = /*@__PURE__*/ BUILD.hotModuleReplacement
-  ? ((window as any).__STENCIL_HOSTREFS__ ||= new WeakMap())
-  : new WeakMap();
-
 /**
  * Given a {@link d.RuntimeRef} remove the corresponding {@link d.HostRef} from
  * the {@link hostRefs} WeakMap.
  *
  * @param ref the runtime ref of interest
  * @returns — true if the element was successfully removed, or false if it was not present.
  */
-export const deleteHostRef = (ref: d.RuntimeRef) => hostRefs.delete(ref);
+export const deleteHostRef = (ref: d.RuntimeRef): boolean => {
```

Comment:
Do we still need this function? I added it to clean up hostRef entries in the weakmap